src/cart/crt_group.c

```diff
@@ -1040,6 +1040,31 @@ crt_grp_lookup_int_grpid(uint64_t int_grpid)
 	return grp_priv;
 }
 
+/* lookup by string subgrp id */
+struct crt_grp_priv *
+crt_grp_lookup_grpid(crt_group_id_t grp_id)
+{
+	struct crt_grp_priv	*grp_priv;
+	bool			found = false;
+
+	D_RWLOCK_RDLOCK(&crt_grp_list_rwlock);
+	d_list_for_each_entry(grp_priv, &crt_grp_list, gp_link) {
+		if (crt_grp_id_identical(grp_priv->gp_pub.cg_grpid,
+					 grp_id)) {
+			found = true;
+			break;
+		}
+	}
+
+	if (found == true)
+		crt_grp_priv_addref(grp_priv);
+	else
+		grp_priv = NULL;
+	D_RWLOCK_UNLOCK(&crt_grp_list_rwlock);
+
+	return (found == true) ? grp_priv : NULL;
```

Comment:
can just return grp_priv here without the check. line 1062 already sets grp_priv to null if found== false

Reply:
Done